wme_junctionangle.user.js

```diff
@@ -502,9 +529,28 @@ function run_ja() {
 					return ja_routing_type.EXIT;
 				}
 			}
+			
+			//wlodek76: FIXING KEEP LEFT/RIGHT regarding to left most segment
+			//WIKI WAZE: When there are more than two segments less than 45.04°, only the left most segment will be KEEP LEFT, all the rest will be KEEP RIGHT
+			
+			//wlodek76: KEEP LEFT/RIGHT overlapping case
+			//WIKI WAZE: If the left most segment is overlapping another segment, it will also be KEEP RIGHT.
+			if (mostleftangle!=null && mostleftangle2!=null) {
+				var overlapped_angle = Math.abs(mostleftangle[0] - mostleftangle2[0]);
+				
+				// If two top most left angles are close < 2 degree they are overlapped.
+				// Method of recognizing overlapped segment by server is unknown for me yet, I took this from WME Validator information about this.
+				// TODO: verify overlapping check on the side of routing server.
+				if (overlapped_angle < 2.0) mostleftangle = null;
+			}
+			
+			if (mostleftangle != null && mostleftangle[1] == s_out_id) {

```

Comment:
These special considerations does not matter from LHT/RHT and they should work in the same way for left and right hand country, because we see the road the same. We get the same turn left/right for LHT and RHT. Of course difference is with exists, in LHT we have left exit and in RHT we have right exit and it is implemented in JAI well and works excellent already.

Example:
Both keep segments are on the right side of road (-8 and -35 degree) and we get correct keep with these patch as on real LiveMap:
https://www.waze.com/pl/editor/?env=row&lon=-0.01150&lat=51.56457&layers=388&zoom=9&segments=121714159
![image](https://cloud.githubusercontent.com/assets/7284315/7219064/e2088534-e68e-11e4-91c2-01a2207a383b.png)

But to make sure we could try to find more examples in UK, Japan or India.